### PR TITLE
Add import null reference exception

### LINKED ISSUE #48891: ArgumentNullException from AbstractAddImportFeatureService`1.GetCollectionExpressionType
**Version Used**: 

```
System.ArgumentNullException: Value cannot be null.
Parameter name: node
      at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetTypeInfoFromNode(SyntaxNode node, CancellationToken cancellationToken)
      at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetTypeInfoCore(SyntaxNode node, CancellationToken cancellationToken)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.GetCollectionExpressionType(SemanticModel semanticModel, ISyntaxFacts syntaxFactsService, SyntaxNode node)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.SymbolReferenceFinder.<GetReferencesForGetEnumeratorAsync>d__32.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.SymbolReferenceFinder.<DoAsync>d__17.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.<FindResultsInAllSymbolsInStartingProjectAsync>d__23.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.<FindResultsAsync>d__22.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.<FindResultsAsync>d__20.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.<GetFixesInCurrentProcessAsync>d__19.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.<GetFixesAsync>d__18.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at Microsoft.CodeAnalysis.Remote.RemoteMissingImportDiscoveryService.<>c__DisplayClass3_0.<<GetFixesAsync>b__0>d.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at System.Threading.Tasks.ValueTask`1.get_Result()
      at Microsoft.CodeAnalysis.Remote.BrokeredServiceBase.<RunServiceAsync>d__11`1.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
      at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)

```

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes #48891

I couldn't get a repro of this, but its a fairly obvious fix. See comments below.

First commit annotates, second commit fixes the bug.

### CODE REVIEW

File: src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs [hunk 1]
```diff
@@ -526,29 +532,39 @@ private static CodeAction TryCreateCodeAction(Document document, AddImportFixDat
...
             };
 
-        private static ITypeSymbol GetAwaitInfo(SemanticModel semanticModel, ISyntaxFacts syntaxFactsService, SyntaxNode node)
+        private static ITypeSymbol? GetAwaitInfo(SemanticModel semanticModel, ISyntaxFacts syntaxFactsService, SyntaxNode node)
         {
             var awaitExpression = FirstAwaitExpressionAncestor(syntaxFactsService, node);
 
             var innerExpression = syntaxFactsService.GetExpressionOfAwaitExpression(awaitExpression);
 
+            if (innerExpression is null)
```
Comment (davidwengier) [inline src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs:541]: This is a bonus fix. Callers to `GetAwaitInfo` already handle when it returns null.

File: src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs [hunk 2]
```diff
@@ -526,29 +532,39 @@ private static CodeAction TryCreateCodeAction(Document document, AddImportFixDat
...
+
             return semanticModel.GetTypeInfo(innerExpression).Type;
         }
 
-        private static ITypeSymbol GetCollectionExpressionType(SemanticModel semanticModel, ISyntaxFacts syntaxFactsService, SyntaxNode node)
+        private static ITypeSymbol? GetCollectionExpressionType(SemanticModel semanticModel, ISyntaxFacts syntaxFactsService, SyntaxNode node)
         {
             var collectionExpression = FirstForeachCollectionExpressionAncestor(syntaxFactsService, node);
 
+            if (collectionExpression is null)
```
Comment (davidwengier) [inline src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs:553]: This is the real fix. Callers to `GetCollectionExpressionType` already handle when it returns null.

**Review by jasonmalinowski — APPROVED** [review]
Fix looks good.

@davidwengier Did we have any heap dumps of the failure?

**Comment by davidwengier** [comment]
> Did we have any heap dumps of the failure?

No, I think @tmat just hit it randomly yesterday.

**Pull request merged**